kitsune/wiki/views.py

```diff
@@ -1445,3 +1473,116 @@ def what_links_here(request, document_slug):
+                for locale in FALLBACK_LOCALES[request.LANGUAGE_CODE]:
+                    if locale in translated_locales:
+                        fallback_locale = locale
+                        break
+                else:
+                    # If above does not match, then we would check if any locale of
+                    # Accepted_Language Header is in the Custom Wiki Locale mappping
+                    for locale in header_locales:
+                        if locale in FALLBACK_LOCALES:
+                            # Check if the document is trnaslated into the fallback locale
+                            for locale in FALLBACK_LOCALES[locale]:
+                                if locale in translated_locales:
+                                    fallback_locale = locale
+                                    break
+                            else:
+                                continue
+                            break
+                    else:
+                        # If all fails, return None as fallback Locale
+                        fallback_locale = None
+
+            # If Custom Wiki Locale Mapping is not available for the locale user requesed, then we
+            # will check if We have any fallback locale available in Custom Wiki locale
+            # mapping for the locales of Accpeted_Launguage Header.
+            else:
+                for locale in header_locales:
+                    if locale in FALLBACK_LOCALES:
+                        # Check if the document is trnaslated into the fallback locale
+                        for locale in FALLBACK_LOCALES[locale]:
+                            if locale in translated_locales:
+                                fallback_locale = locale
+                                break
+                        else:
+                            continue
+                        break
+                else:
+                    # If all fails, return None as fallback Locale
+                    fallback_locale = None
+    return fallback_locale
+
+
+# Import from django.utils.translation.trans_real and changed as below
+#  * Removed lower() from piece to get the locale name as needed
+#  * As we need only the locale name, added code to get only locale name
+def parse_accept_lang_header(lang_string):
+    """
+    Parses the lang_string, which is the body of an HTTP Accept-Language
+    header, and returns a list of lang, ordered by 'q' values.
+    Any format errors in lang_string results in an empty list being returned.
+    """
+    accept_language_re = re.compile(r'''
+        ([A-Za-z]{1,8}(?:-[A-Za-z0-9]{1,8})*|\*)      # "en", "en-au", "x-y-z", "es-419", "*"
+        (?:\s*;\s*q=(0(?:\.\d{,3})?|1(?:.0{,3})?))?   # Optional "q=1.00", "q=0.8"
+        (?:\s*,\s*|$)                                 # Multiple accepts per header.
+        ''', re.VERBOSE)
+    result = []
+    pieces = accept_language_re.split(lang_string)
+    print pieces   # Changed here. Removed lower()
```

Comment:
This is necessary because I have changed this. According to change in Open source code, need to mention the change log. :smile: